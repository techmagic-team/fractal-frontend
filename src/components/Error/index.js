import React from 'react';
import PropTypes from 'prop-types';

const Error = ({ errorMessage }) => (
  <div className="alert alert-danger" role="alert">
    Request failed{errorMessage ? `: ${errorMessage}` : ''}
  </div>
);

Error.propTypes = {
  errorMessage: PropTypes.string.isRequired,
};

export default Error;
