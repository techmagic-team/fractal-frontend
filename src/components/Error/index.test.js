import React from 'react';
import { create } from 'react-test-renderer';

import Error from './index';

describe('Error', () => {
  it('should match the snapshot without errorMessage', () => {
    const errorMessage = '';
    const component = create(<Error errorMessage={errorMessage} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot with errorMessage', () => {
    const errorMessage = 'Network error';
    const component = create(<Error errorMessage={errorMessage} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
