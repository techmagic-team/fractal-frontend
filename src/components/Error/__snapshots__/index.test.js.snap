// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match the snapshot with errorMessage 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  Request failed
  : Network error
</div>
`;

exports[`Error should match the snapshot without errorMessage 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  Request failed
  
</div>
`;
